src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -1,56 +1,49 @@
 package org.skypro.skyshop.basket;
 
-import org.skypro.skyshop.product.DiscountedProduct;
 import org.skypro.skyshop.product.Product;
-import org.skypro.skyshop.product.SimpleProduct;
+
+import java.util.*;
 
 public class ProductBasket {
-    private Product[] products = new Product[5];
+    private HashMap<String, List<Product>> products;
```

Comment:
Используемы максимально возможный уровень абстракции

```suggestion
    private Map<String, List<Product>> products;
```